x-pack/legacy/plugins/lens/server/usage/collectors.ts

```diff
@@ -21,25 +21,124 @@ export function registerLensUsageCollector(
     type: 'lens',
-    fetch: async (callCluster: CallCluster): Promise<LensUsage> => {
+    fetch: async (): Promise<LensUsage> => {
       try {
-        return getVisualizationCounts(callCluster, plugins.config);
+        const docs = await getLatestTaskState(plugins.server);
+        // get the accumulated state from the recurring task
+        const state: LensTelemetryState = get(docs, '[0].state');
+
+        const dates = Object.keys(state.byDate || {}).map(dateStr => parseInt(dateStr, 10));
+        const suggestionDates = Object.keys(state.suggestionsByDate || {}).map(dateStr =>
+          parseInt(dateStr, 10)
+        );
+
+        const eventsLast30: Record<string, number> = {};
+        const eventsLast90: Record<string, number> = {};
+        const suggestionsLast30: Record<string, number> = {};
+        const suggestionsLast90: Record<string, number> = {};
+
+        const last30 = moment()
+          .subtract(30, 'days')
+          .unix();
+        const last90 = moment()
+          .subtract(90, 'days')
+          .unix();
+
+        dates.forEach(date => {
+          if (date >= last30) {
+            addEvents(eventsLast30, state.byDate[date]);
+            addEvents(eventsLast90, state.byDate[date]);
+          } else if (date > last90) {
+            addEvents(eventsLast90, state.byDate[date]);
+          }
+        });
+
+        suggestionDates.forEach(date => {
+          if (date >= last30) {
+            addEvents(suggestionsLast30, state.suggestionsByDate[date]);
+            addEvents(suggestionsLast90, state.suggestionsByDate[date]);
+          } else if (date > last90) {
+            addEvents(suggestionsLast90, state.suggestionsByDate[date]);
+          }
+        });
+
+        return {
+          ...state.saved,
+          events_30_days: eventsLast30,
+          events_90_days: eventsLast90,
+          suggestion_events_30_days: suggestionsLast30,
+          suggestion_events_90_days: suggestionsLast90,
+        };
       } catch (err) {
         return {
-          saved_total: 0,
-          saved_last_30_days: 0,
-          saved_last_90_days: 0,
-          visualization_types_overall: {},
-          visualization_types_last_30_days: {},
-          visualization_types_last_90_days: {},
+          saved_overall_total: 0,
+          saved_30_days_total: 0,
+          saved_90_days_total: 0,
+          saved_overall: {},
+          saved_30_days: {},
+          saved_90_days: {},
+
+          events_30_days: {},
+          events_90_days: {},
+          suggestion_events_30_days: {},
+          suggestion_events_90_days: {},
         };
       }
     },
-    isReady: () => true,
+    isReady: () => isCollectorReady,
   });
   plugins.usage.collectorSet.register(lensUsageCollector);
 }
+
+function addEvents(prevEvents: Record<string, number>, newEvents: Record<string, number>) {
+  Object.keys(newEvents).forEach(key => {
+    prevEvents[key] = prevEvents[key] || 0 + newEvents[key];
```

Comment:
I think this would benefit from parenthesis. It *seems* wrong to me, but I may be misunderstanding the intent. Is the intent to add `newEvents[key]` to `prevEvents[key]`? If so, I don't think this is what will happen here.

```js
// This is what is happening
prevEvents[key] = prevEvents[key] || (0 + newEvents[key]);

// I think this is what you want...
prevEvents[key] = (prevEvents[key] || 0) + newEvents[key];

```

Reply:
This is actually a linting thing- it doesn't let me add parentheses here

Reply:
I'm not sure why lint won't let you put parenthesis there, but I think you should, and should add a lint exception comment.

Reply:
I must have been wrong here. I did at one point see weird issues where the linter would reformat the code, but maybe I was not doing this the right way at the time